### ISSUE TITLE
better detection in Check if database is already created

### ISSUE THREAD

**Issue by Rendanic** [oravirt/ansible-oracle#17]
The current method has issues in RAC environments. 
I think the following logic should work:
- no change for current logic with register: checkdbexist
  There is no need to change the other tasks for the new logic.
- add another when to current task
  The existing task gets another when clause and only applies to installations without Oracle Restart / GI.
  Also change the search for the SID to '^item.oracle_db_name:oracle_home:'. 
  => We really find the right entry in oratab for the ORACLE_SID
- add a new task like the existing Check if database is already created
  This task is only for Oracle Restart / GI. We could get the state of the database with srvctl status database -d <item.oracle_db_name> and return '1' when a database was found on the system.
  This is compatible against the current logic and is easy to add without changing a lot of tasks.

What do you think about this idea?

**Comment by Rendanic** [comment]
Here is an example how it works with Clusterware/Oracle Restart. It doesn't look like a really nice solution but it is working in the current `oradb-create` role. One problem was, that the role `oradb-create` doesn't know where GI/Restart is installed. I get that information from `/etc/oracle/olr.loc` which is existing in 11.2+.

```
  - name: Check if database is already created (2)
    shell: ". /etc/oracle/olr.loc ;${crs_home}/bin/srvctl config database -d {{ item.oracle_db_name }} > /dev/null ; if [ $? -eq 0 ] ; then echo '1' ; else echo '0' ; fi"
    with_items: oracle_databases
    when: item.oracle_db_name is defined and oracle_install_option_gi in ('CRS_CONFIG','HA_CONFIG')
    tags:
    - checkifdbexists
    register: checkdbexist

```

**Comment by oravirt** [comment]
What are the problems you had with the detection with a RAC config? I'm just wondering because I've never had issues with the way the detection has worked so far.

The only thing that I do differently in an RAC-config, is that for each node I also add the instance-name using ansibles 'lineinfile' module, after the database is created.
So after an successful installation oratab will look like this on the 'first' node:

orcl:/u01/app/oracle/11.2.0.4/db1:N        # line added by agent
orcl1:/u01/app/oracle/11.2.0.4/db1:N       <--- This line is added by the lineinfile module

That specific task 'Update oratab' (using lineinfile) needs to be adjusted, but even adding that line doesn't interfere with the initial detection of the DB. I only ever look for the dbname, never the instance.

Changing the grep to your suggestion is absolutely fine, but I'd rather not use different methods depending on config unless there is a real need for it.

**Comment by Rendanic** [comment]
You have to add every possible SID in every oratab file in policy managed installation. Otherwise you'll miss some entries.
That's why I started building scripts, who doesn't rely on special ORACLE_SIDs in oratab.

There are also a couple of bugs with adding entries in oratab and loosing them with the next start of clusterware.

I had really stupid vonfigurations at customers, which leads to bad results, when the ORACLE_SID is detected like ansible-oracle is doing it. I prefer real clean solutions, because it is hard to test all possible values for ORACLE_SIDs.
This is absolutly fine for known envirenments and admins who knows how it works, but the reality at customers is somtimes a little bit different...

**Comment by oravirt** [comment]
Ok.
Then I think we should consolidate into one check that dynamically checks if the host runs GI and in that case uses srvctl to see if the db exists or not, and if the host doesn't run GI falls back to the old way of checking (grep'ing oratab).

Make it a shellscript that is run either via the script module (which gets the dbname + oracle_home passed in as variables) or as a template that gets executed.

**Comment by oravirt** [comment]
This is addressed by 174e8e03. It checks if GI is installed, and if so uses `srvctl config database` to check if the DB is present or not. 
If GI is not present, it checks `/etc/oratab`
